### ISSUE TITLE
vomnibarFrame and vimiumHUDFrame visible on new tab

### ISSUE THREAD

**Issue by mrsean2k** [philc/vimium#1817]
OSX 10.10.5 / Chrome 46.0.2490.22 beta (64-bit)

When a new empty tab is opened, the two frames mentioned in the title are visible as in the attached screenshot.

These don't appear anywhere except the initial empty tab and don't seem to have any adverse effect otherwise
Started within the last week - no idea which specific Chrome update triggered it, sorry.

![developer_tools_-_https___www_google_co_uk___chrome_newtab-serviceworker_js_and_new_tab_and_xara_viewer_for_osx_-_google_search_and_react_native_debugger](https://cloud.githubusercontent.com/assets/3684539/9809491/8acb9916-5860-11e5-907a-b2364d3521a1.png)

**Comment by smblott-github** [comment]
Thanks, @mrsean2k.  For sure this started with 1.52 (released a couple of days ago).

@mrmr1993 is our expert for this type of thing.
Any ideas, @mrmr1993?

**Comment by mrmr1993** [comment]
I can repro that Chrome now runs content scripts on the New Tab Page, but not that the frames aren't having their styles supplied (although I did notice this once last week running under the Linux Chrome `unstable` channel). My thoughts:
* Extensions run on the NTP now because the actual URL is now `https://www.google.co.uk/_/chrome/newtab` instead of an internal one.
* I get an warning message `Resource interpreted as Stylesheet but transferred with MIME type text/plain: "chrome-extension://dbepggeogbaibhgnhhndojpepiihcmeb/content_scripts/vimium.css"`, which seems suspiciously like the kind of thing that might have caused this.
  - @mrsean2k, can you see if you have any errors in the developer tools console?
* I can use Vimium fully in the NTP now. Apart from the UI not being styled correctly, does everything still work when the frames are visible?

**Comment by dbanck** [comment]
Same visual errors for me with OS X and Arch Linux  / 46.0.2490.22 beta (64-bit).

These are the errors from my developer console.
![Error Console](http://i.imgur.com/L1ZKQKw.png)

Hotkeys like 't', 'f', '?' are working fine on NTP

**Comment by philc** [comment]
Content scripts running on NTP? Exciting!

**Comment by mrmr1993** [comment]
Bizarre, the CSS file won't load. If you visit the URL for it manually (`chrome-extension://dbepggeogbaibhgnhhndojpepiihcmeb/content_scripts/vimium.css`), does it come up?

I'm presuming that reloading the extension (disable and re-enable Vimium in chrome://extensions or close Chrome and open it again) should fix the error, as I expect the cause is either:
* `vimium.css` hasn't been loaded into Chrome from the extension properly, or
* the `web_accessible_resources` entry isn't being respected for whatever reason.

**Comment by smblott-github** [comment]
> If you visit the URL for it manually (chrome-extension://dbepggeogbaibhgnhhndojpepiihcmeb/content_scripts/vimium.css), does it come up?

(I have this reproduced.)

Yes.  You see the CSS.

The problem seems to be happening here (see message about "promise being rejected, above)...

![snapshot](https://cloud.githubusercontent.com/assets/2641335/9812088/f2433e98-5872-11e5-9ceb-166178814f23.png)

**Comment by mrmr1993** [comment]
That's where the error's being thrown, but that's not our code. And the error message tells us exactly what's going on: the page can't load the file. Which would seem to be a Chromium issue. 

Does reloading the extension/restarting the browser help for anyone?

**Comment by dbanck** [comment]
I think the NTP is using a [service worker](https://www.chromium.org/blink/serviceworker) (`_/chrome/newtab-serviceworker.js`) for loading extensions.

The URL scheme `chrome-extension` doesn't seem to work with service workers (yet): https://code.google.com/p/chromium/issues/detail?id=517181

**Comment by smblott-github** [comment]
> Does reloading the extension/restarting the browser help for anyone?

No.

**Comment by smblott-github** [comment]
A quick-fix might be to disable Vimium on the NTP until we figure out what's going on...

![snapshot](https://cloud.githubusercontent.com/assets/2641335/9812494/a7ce8982-5875-11e5-82e3-0be557311b86.png)

Some stuff works, but other stuff doesn't; e.g. `Escape` doesn't get you out of the Vomnibar.

**Comment by mrsean2k** [comment]
Sorry, intervening dogwalk:

1) restarting doesn't seem to make any difference for me

2) my console log is here

<pre>
Fetch API cannot load chrome-extension://pioclpoplcdbaefihamjohnefbikjilc/content/require/topFrame.js. URL scheme "chrome-extension" is not supported.
The FetchEvent for "chrome-extension://pioclpoplcdbaefihamjohnefbikjilc/content/require/topFrame.js" resulted in a network error response: the promise was rejected.
Uncaught (in promise) TypeError: Failed to fetch(…)
Fetch API cannot load chrome-extension://pioclpoplcdbaefihamjohnefbikjilc/content/Clipper.js. URL scheme "chrome-extension" is not supported.
The FetchEvent for "chrome-extension://pioclpoplcdbaefihamjohnefbikjilc/content/Clipper.js" resulted in a network error response: the promise was rejected.
Uncaught (in promise) TypeError: Failed to fetch(…)
Fetch API cannot load chrome-extension://pioclpoplcdbaefihamjohnefbikjilc/content/ContentPreview.js. URL scheme "chrome-extension" is not supported.
The FetchEvent for "chrome-extension://pioclpoplcdbaefihamjohnefbikjilc/content/ContentPreview.js" resulted in a network error response: the promise was rejected.
Uncaught (in promise) TypeError: Failed to fetch(…)
Fetch API cannot load chrome-extension://pioclpoplcdbaefihamjohnefbikjilc/content/Coordinator.js. URL scheme "chrome-extension" is not supported.
The FetchEvent for "chrome-extension://pioclpoplcdbaefihamjohnefbikjilc/content/Coordinator.js" resulted in a network error response: the promise was rejected.
Uncaught (in promise) TypeError: Failed to fetch(…)
Fetch API cannot load chrome-extension://pioclpoplcdbaefihamjohnefbikjilc/js/GlobalUtils.js. URL scheme "chrome-extension" is not supported.
The FetchEvent for "chrome-extension://pioclpoplcdbaefihamjohnefbikjilc/js/GlobalUtils.js" resulted in a network error response: the promise was rejected.
Uncaught (in promise) TypeError: Failed to fetch(…)
Fetch API cannot load chrome-extension://pioclpoplcdbaefihamjohnefbikjilc/content/tooltips/CustomTooltipEligibility.js. URL scheme "chrome-extension" is not supported.
The FetchEvent for "chrome-extension://pioclpoplcdbaefihamjohnefbikjilc/content/tooltips/CustomTooltipEligibility.js" resulted in a network error response: the promise was rejected.
Uncaught (in promise) TypeError: Failed to fetch(…)
Failed to load resource: net::ERR_FAILED
Failed to load resource: net::ERR_FAILED
Failed to load resource: net::ERR_FAILED
Failed to load resource: net::ERR_FAILED
Failed to load resource: net::ERR_FAILED
Failed to load resource: net::ERR_FAILED
SW registered
Fetch API cannot load chrome-extension://dbepggeogbaibhgnhhndojpepiihcmeb/content_scripts/vimium.css. URL scheme "chrome-extension" is not supported.
The FetchEvent for "chrome-extension://dbepggeogbaibhgnhhndojpepiihcmeb/content_scripts/vimium.css" resulted in a network error response: the promise was rejected.
Uncaught (in promise) TypeError: Failed to fetch(…)
Failed to load resource: net::ERR_FAILED
Failed to load resource: net::ERR_FAILED
Failed to load resource: net::ERR_FAILED
Failed to load resource: net::ERR_FAILED

</pre>


but it look as if you're past all that now :)

**Comment by smblott-github** [comment]
(Looks like Evernote is having a hard time too! -- Although I guess that matters less.)

**Comment by mrmr1993** [comment]
It's [this Chromium issue](https://code.google.com/p/chromium/issues/detail?id=466876). The breaking change is not present in stable (45.0.2454.85), present in beta (46.0.2490.22) and fixed in dev (47.0.2503.0). I presume you're all running beta.

> A quick-fix might be to disable Vimium on the NTPA quick-fix might be to disable Vimium on the NTP

That should work for the time being, although it looks the issue will fix itself at some point in the next few weeks.

**Comment by innaterebel** [comment]
I have no programming knowledge, but my layman guess why extensions now work on NTP is because the current NTP design is a partial web page, still has an internal URL and has local installed theme integrated then with some elements loaded from web like doodles.

And in my experience the first NTP opened following browser initiating is a more local one with no extension works on it, apparently for speed reason or just has higher loading priority than extensions during browser initiation.

In short, the whole 'extensions now work on NTP' might just not be intended but a side-effect and you couldn't get any guarantee from it.

Again, this is purely my stranger guessing so apologize for interruption if I am totally wrong.

**Comment by innaterebel** [comment]
Forgot to mention, for me some keys (like `closeTabs` ones ,I just didn't try many) has worked on afterward opened NTP for some long time, not only after 1.52. Although 1.52 may make it more apparent.
